pages/search_page.py

```diff
@@ -10,16 +10,19 @@ class SearchPage(BasePage):
     __CLOSE_BUTTON_FOR_CONFIRMATION_POPUP = (By.XPATH, '//*[@id="layer_cart"]/div[1]/div[1]/span')
     __SEARCH_RESULT_ITEM = (By.XPATH, '//*[@id="center_column"]/ul/li[1]/div')
     __ADD_BUTTON_FOR_SEARCH_RESULT_ITEM = (By.XPATH, '//*[@id="center_column"]/ul/li[1]/div/div[2]/div[2]')
-    __5TH_PRODUCT = (By.XPATH, '//*[@id="center_column"]/ul/li[5]')
+
+    # dynamic xpaths
+    __PRODUCT = (By.XPATH, '//*[@id="center_column"]/ul/li[{index}]')
+
+    def check_if_search_block_contains_element(self, expected_product_quantity: int) -> bool:
+        element_locator = (By.XPATH, self.__PRODUCT.format(index=expected_product_quantity))
+        self.if_element_present(by_locator=element_locator)
 
     def search_page_title(self):
-        return self.get_element(by_locator=self.__SEARCH_PAGE_TITLE)
+        self.if_element_present(by_locator=self.__SEARCH_PAGE_TITLE)
 
     def search_result_item(self):
-        return self.get_element(by_locator=self.__SEARCH_RESULT_ITEM)
-
-    def search_5th_product(self):
-        return self.get_element(by_locator=self.__5TH_PRODUCT)
+        self.if_element_present(by_locator=self.__SEARCH_RESULT_ITEM)
 
     def add_item_to_cart(self):
```

Comment:
a small remark about naming. `add_item_to_cart` is a great name for a method that will take an `item` as an argument:
`def add_item_to_cart(item_name: str) -> None:`

if you want to add e.g. a "fixed item" then probably better name would be `add_first_result_item_to_cart`